src/Symfony/Component/HttpFoundation/Tests/RequestTest.php

```diff
@@ -326,6 +326,23 @@ public function testGetMimeTypeFromFormat($format, $mimeTypes)
         }
     }
 
+    /**
+     * @dataProvider getFormatToMimeTypeMapProviderForAccept
+     */
+    public function testGetMimeTypeFromFormatWithAccept($format, $mimeTypes, $accepts)
+    {
+        if (null !== $format) {
+            $request = new Request();
+            $request->setFormat($format, $mimeTypes);
+
+            $request->headers->set('Accept', $accepts);
+            // BC behavior.
+            // If accept is inside of the $format mime type array, return it, otherwise behave as you currently do.
+            $return = $accepts ?: $mimeTypes[0];
```

Comment:
Currently, Request::getMimeTypes($format) returns the first element in the mimetypes array, if $format is set as a key in it.

All we're doing here is saying, if $format is indeed set, but also the accept header is set, check to see if the element exists first, and return it, if not, then go ahead and return the first element. This should be BC.